### ISSUE TITLE
`EventIcompatibleServer` is sent when an unexpected sse error is caught

### ISSUE THREAD

**Issue by vxgmichel** [Scille/parsec-cloud#7474]
While testing, the front team received a confusing `IncompatibleServer` error. 

It likely originated from this catch-all in `handle_sse_error`:

https://github.com/Scille/parsec-cloud/blob/5da5a48b635e1c02e7486a0411a729540e1f83c8/libparsec/crates/client/src/monitors/connection.rs#L165-L170

@Max-7 is adding more debug information to the front to identify the specific error that occurred.
@touilleMan is this the expected behavior? `IncompatibleServer` should probably be reserved to API mismatch versions IMO.

**Comment by vxgmichel** [comment]
More precisely:
```
[2024-06-07 11:44:02.815] [warn]  IncompatibleServerEvent: Server is incompatible due to an unexpected error: Unexpected response status 502 Bad Gateway
```

**Comment by vxgmichel** [comment]
Also shouldn't this:
https://github.com/Scille/parsec-cloud/blob/3f19d2d365f8385acce9bd8d515adf5fea3243ff/libparsec/crates/client/src/monitors/connection.rs#L238-L241
be:
```rust
                Err(err) => {
                    if handle_sse_error(&mut state, &event_bus, err).is_break() {
                        break;
                    }
                }
```
to allow for a new connection after a connexion loss?

**Comment by touilleMan** [comment]
> Also shouldn't this:
> [...]
>  be:
> [...]
> to allow for a new connection after a connexion loss?

`handle_sse_error` returns a `ControlFlow` that tells the caller whether or not it should allow for a new connection after connexion loss.
The idea is `ConnectionError::NoResponse` (so HTTP status 502/503/504, or not a broken connection with the server) should indeed lead to retry, but all other kind of response shouldn't to avoid polling the server pointlessly (if the server has answered gibberish, there is no reason it will change anytime soon...)

**Comment by vxgmichel** [comment]
#7550 does not fix the issue, the following code also has to be modified:
https://github.com/Scille/parsec-cloud/blob/fc3f00f2caeeb4d539ee30d9676c1c62b494c6d6/libparsec/crates/client_connection/src/authenticated_cmds/sse.rs#L157-L158

**Comment by vxgmichel** [comment]
Also we should probably reconsider the fact that any `ConnectionError` other than:
- `NoResponse`
- `ExpiredOrganization`
- `RevokedUser`
- `UnsupportedApiVersion`

leads to an `IncompatibleServerEvent`. 

In particular, do we want all `InvalidResponseStatus` to lead to permanently offline status? Are we sure we can be exhaustive with the HTTP status return by the web server? `502/503/504` might not be the only ones.

> The idea is ConnectionError::NoResponse (so HTTP status 502/503/504, or not a broken connection with the server) should indeed lead to retry, but all other kind of response shouldn't to avoid polling the server pointlessly (if the server has answered gibberish, there is no reason it will change anytime soon...)

I would challenge this as well. How is polling the server a bad thing? Maybe the server is in a temporary invalid state causing an expected error, but it's going to get a back to valid state and some point so it makes sense for the client to poll until this happens.

I would argue the opposite: it only makes sense not to poll if the server explicitly replies that the API version are incompatible. In this case, the only fix is to update the client (since servers are not meant to be downgraded).

**Comment by FirelightFlagboy** [comment]
Fixed in #7572
